cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -108,105 +108,156 @@ Given(/1st party app invokes the (?:'(.+)' )?API to '(.+)'$/, async (sdk, key) =
+              // Sending message to 3rd party app.
+              cy.sendMessagetoApp(requestTopic, responseTopic, parsedIntent).then((result) => {
+                if (!Cypress.env('isRpcOnlyValidation')) {
+                  if (result === CONSTANTS.NO_RESPONSE) {
+                    assert(false, CONSTANTS.NO_MATCHED_RESPONSE);
+                  }
+                  result = JSON.parse(result);
+                  cy.updateResponseForFCS(method, params, result).then((updatedResponse) => {
+                    // Create a deep copy to avoid reference mutation
+                    const dataToBeCensored = _.cloneDeep(result);
+
+                    // Call the 'censorData' command to hide sensitive data
+                    cy.censorData(method, dataToBeCensored).then((maskedResult) => {
+                      const responseType = result.error ? CONSTANTS.ERROR : CONSTANTS.RESULT;
+                      fireLog.info(
+                        `Response from app: ${appId}, device:  ${deviceIdentifier || UTILS.getEnvVariable(CONSTANTS.DEVICE_MAC)} - ${JSON.stringify(maskedResult[responseType])}`
+                      );
+                    });
+
+                    // If method and params are not supported setting isScenarioExempted as true for further validation.
+                    if (UTILS.isScenarioExempted(method, param)) {
+                      Cypress.env(CONSTANTS.IS_SCENARIO_EXEMPTED, true);
+                    }
 
-              // Creating object with method name, params and response etc and storing it in a global list for further validation.
-              const apiAppObject = new apiObject(
-                method,
-                param,
-                context,
-                updatedResponse,
-                expected,
-                appId
-              );
-              UTILS.getEnvVariable(CONSTANTS.GLOBAL_API_OBJECT_LIST).push(apiAppObject);
+                    // Creating object with method name, params and response etc and storing it in a global list for further validation.
+                    const apiAppObject = new apiObject(
+                      method,
+                      param,
+                      context,
+                      updatedResponse,
+                      expected,
+                      appId
+                    );
+                    UTILS.getEnvVariable(CONSTANTS.GLOBAL_API_OBJECT_LIST).push(apiAppObject);
+                  });
+                }
+              });
             });
-          }
+          });
         });
+      })
+      .then(() => {
+        if (deviceIdentifier && deviceName !== CONSTANTS.DEVICE1) {
+          const publishMessage = UTILS.createIntentMessage(
+            CONSTANTS.TASK.CALLMETHOD,
+            { method: 'Lifecycle.close', methodParams: { reason: 'error' } },
+            { communicationMode: UTILS.getCommunicationMode() }
+          );
+          cy.sendMessagetoApp(requestTopic, responseTopic, publishMessage);
+        }
```

Comment:
what is the need for this logic ?

Reply:
above logic is to close the app when we passe device identifier apart from device1